### PR TITLE
Rs network functions

### PR DESCRIPTION
**Pull request by Kraemii**
**What**:
This PR is base on https://github.com/greenbone/openvas-scanner/pull/1649
Jira: SC-1120

Implementing the following network functions:
- this_host_name
- get_mtu
- this_host
- islocalhost
- islocalnet
- get_host_ip
- scanner_add_port

for testing I used a minimal NASL script:
```
display(<network_function>);
```
and execute `scannerctl execute script -t <target> test.nasl`

### CODE REVIEW

**Review by jjnicola — CHANGES_REQUESTED** [review]

File: rust/nasl-builtin-network/src/network.rs [hunk 1]
```diff
@@ -45,9 +45,58 @@ fn nasl_islocalhost(context: &Context) -> Result<bool, FunctionErrorKind> {
...
+    let dst_str = dst.to_string();
+    let src_str = src.to_string();
+
+    if dst.is_ipv4() {
+        let dst_parts: Vec<&str> = dst_str.split('.').collect();
+        let src_parts: Vec<&str> = src_str.split('.').collect();
+        let netmask_parts: Vec<&str> = netmask_str.split('.').collect();
+
+        for i in 0..4 {
+            if netmask_parts[i] == "0" {
```
[the remark this answers is not in the document]
Comment (jjnicola) [reply rust/nasl-builtin-network/src/network.rs:76]: Same for IPv6

[the remark this answers is not in the document]
Comment (Kraemii) [reply rust/nasl-builtin-network/src/network.rs]: Should be fixed now

**Review by Tehforsch — CHANGES_REQUESTED** [review]
Looks good, just had a few comments.

File: rust/nasl-builtin-network/src/lib.rs [hunk 2]
```diff
@@ -137,3 +148,13 @@ pub fn get_pos_port(r: &Register) -> Result<u16, FunctionErrorKind> {
...
+
+pub fn verify_port(port: i64) -> Result<u16, FunctionErrorKind> {
+    if !(0..=65535).contains(&port) {
+        return Err(FunctionErrorKind::WrongArgument(format!(
+            "{} is not a valid port number",
+            port
+        )));
+    }
+    Ok(port as u16)
+}
```
[the remark this answers is not in the document]
Comment (Kraemii) [reply rust/nasl-builtin-network/src/lib.rs:160]: I guess ports are only relevant for the network and raw ip functions, at least regarding NASL

File: rust/nasl-builtin-network/src/network_utils.rs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    unsafe {
+        libc::freeifaddrs(interfaces);
+    }
+    Err(FunctionErrorKind::Diagnostic(
+        "No route to destination".to_string(),
+        None,
+    ))
+}
```
[the remark this answers is not in the document]
Comment (Kraemii) [reply rust/nasl-builtin-network/src/network_utils.rs]: I did simplify the unsafe blocks a little bit, to have a better view and wrote a small comment, why this is unsafe

File: rust/nasl-builtin-network/src/network_utils.rs [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+/// Convert a string in a IpAddr
+pub fn ipstr2ipaddr(ip_addr: &str) -> Result<IpAddr, FunctionErrorKind> {
+    match IpAddr::from_str(ip_addr) {
+        Ok(ip) => Ok(ip),
+        Err(_) => Err(FunctionErrorKind::Diagnostic(
+            format!("Invalid IP address ({})", ip_addr),
+            Some(NaslValue::Null),
+        )),
+    }
+}
```
[the remark this answers is not in the document]
Comment (Kraemii) [reply rust/nasl-builtin-network/src/network_utils.rs:25]: Yes, might be a good idea

**Review by jjnicola — APPROVED** [review]

**Pull request merged**